src/EmblemRenderer.vala

```diff
@@ -0,0 +1,108 @@
+/***
+    Copyright (c) 2021 Elementary, Inc <https://elementary.io>
+
+    Pantheon Files is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation; either version 2 of the
+    License, or (at your option) any later version.
+
+    Pantheon Files is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    General Public License for more details.
+
+    You should have received a copy of the GNU General Public
+    License along with this program; see the file COPYING.  If not,
+    write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+    Boston, MA 02110-1335 USA.
+
+    Author(s):  Jeremy Wootten <jeremy@elementaryos.org>
+
+***/
+
+namespace Marlin {
+    public class EmblemRenderer : Gtk.CellRenderer {
+        private GOF.File? _file;
+        public GOF.File? file {
+            get {
+                return _file;
+            }
+            set {
+                _file = value;
+            }
+        }
```

Comment:
Is there any difference between this and:
```suggestion
        public GOF.File? file { get; set; }
```

Reply:
Good point